src/channel_state.ts

```diff
@@ -519,10 +600,10 @@ export class ChannelState<
     // for empty list just concat and return
     if (messageArr.length === 0) return messageArr.concat(message);
 
-    const messageTime = message.created_at.getTime();
+    const messageTime = message[sortBy]?.getTime() || 0;
```

Comment:
did you try casting the type? `(message[sortBy] as ImmutableDate).getTime()` or `(message[sortBy] as Date).getTime()`

Reply:
As I said, the types works fine now. It was an issue in `pinned_at` definition when I was trying to do this originally

Reply:
great, it's generally good practice to avoid adding unnecessary js code to satisfy TS compiler, type casting is cleaner and faster here, especially since this function is a performance bottleneck

Reply:
Is this better? https://github.com/GetStream/stream-chat-js/pull/578/commits/2d2bea43f7cb2959b4141fc752446d1e2d3be2a4

Reply:
Looks great!